### PR TITLE
drivers: clock_control: Update drivers to use devicetree Kconfig symbol

### PR DESCRIPTION
**Pull request by galak**
Update clock_control drivers to use DT_HAS_<compat>_ENABLED Kconfig
symbol to expose the driver and enable it by default based on
devicetree.
    
We remove 'depend on' Kconfig for symbols that would be implied by
the devicetree node existing.

### CODE REVIEW

File: drivers/clock_control/clock_control_mcux_mcg.c [hunk 1]
```diff
@@ -37,9 +37,11 @@ static int mcux_mcg_get_rate(const struct device *dev,
 	clock_name_t clock_name;
 
 	switch ((uint32_t) sub_system) {
+#ifdef kCLOCK_McgFixedFreqClk
```
Comment (mmahadevan108) [inline drivers/clock_control/clock_control_mcux_mcg.c:40]: Why is this needed?

Comment (galak) [reply drivers/clock_control/clock_control_mcux_mcg.c:40]: Because KW40W/KW41Z don't support it so if you build mcg on those platforms it fails to build.

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
@nordic-krch ping

**Review by nordic-krch — APPROVED** [review]
